doc/glossary/index.rst

```diff
@@ -31,3 +31,374 @@ Glossary
+
+   Custom User Attributes
+        The table ``“customuserattribute”`` is used to store additional, custom attributes for users.
+
+        A user is identified by the user_id, the resolver_id and the realm_id.
+        The additional attributes are stored in Key and Value.
+        The Type can hold extra information like e.g. an encrypted value / password.
+
+        .. note:: Since the users are external, i.e. no objects in this database, there is not
+            logic reference on a database level. Since users could be deleted from user stores without
+            privacyIDEA realizing that, this table could pile up with remnants of attributes.
+
+   Scope
+        A scope is the area, where a policy is meant for.
+        This can be values like:
+
+        * ADMIN = 'admin'
+
+        * AUDIT = 'audit'
+
+        * AUTH = 'authentication'
+
+        * AUTHZ = 'authorization'
+
+        * ENROLL = 'enrollment'
+
+        * REGISTER = 'register'
+
+        * USER = 'user'
+
+        * WEBUI = 'webui'
+
+        scope takes only one value.
+
+   Realms
+        Users need to be in realms to have tokens assigned. A user, who is not
+        member of a realm can not have a token assigned and can not authenticate.
+
+        You can combine several different UserIdResolvers (see :ref:`useridresolvers`)
+        into a realm.
+
+        The system knows one default realm. Users within this default realm can
+        authenticate with their username.
+
+        Users in realms, that are not the default realm, need to be additionally identified.
+        Therefore the users need to authenticate with their username and the realm like this::
+
+            user@realm
+
+   Events
+        Each **API call** is an **event** and you can bind arbitrary actions to each
+        event as you like. You can bind several actions to one event. These actions are executed
+        in the order of the priority one after another.
+
+        .. Note:: An action, that is triggered by an event can not trigger a new action. Only **events** (API calls)
+           can trigger actions. E.g. if you are using the :ref:`tokenhandler` to create a new token, the creation
+           of the token is an *action*, not an *event*. This means this creation of the token can *not* trigger a new
+           action. For more complex actions, you might need to look into the :ref:`scripthandler`.
+
+        Internally events are marked by a decorator "event" with an *event identifier*.
+        At the moment not all events might be tagged. Please drop us a note to tag
+        all further API calls.
+
+        .. figure:: /eventhandler/event-list.png
+           :width: 500
+
+           *An action is bound to the event* token_init.
+
+   Radius Attribute Mapping
+        The Radius plugin can use information from the ``detail`` section
+        (see :ref:`rest_validate`) of the
+        privacyIDEA response to map these values to arbitrary RADIUS Attribute-Value
+        pairs.
```

Comment:
https://www.youtube.com/watch?v=uERhuCLxz0o

Reply:
done in f4e110ab89c11f9bc7603cc003a4c8d438ac1665.